arviz/stats/diagnostics.py

```diff
@@ -510,13 +514,22 @@ def geweke(ary, first=0.1, last=0.5, intervals=20):
         last_slice = ary[int(end - last * (end - start)) :]
 
         z_score = first_slice.mean() - last_slice.mean()
-        z_score /= np.sqrt(first_slice.var() + last_slice.var())
+        if _numba_flag:
```

Comment:
My next challenge for you is to get rid of all these if else statements.

There might be a pattern where we can do something like

```
z_score = _numba(f1, f2, args)
```

This way all the numba specific logic is contained in the Numba class and all the math logic is left here. What do you think?

Reply:
Sure.